packages/payment-processor/src/payment/erc20-escrow-payment.ts

```diff
@@ -0,0 +1,396 @@
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function completeEmergencyClaim(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Web3Provider | Signer = getProvider(),
+  overrides?: ITransactionOverrides,
+): Promise<ContractTransaction> {
+  const encodedTx = encodeCompleteEmergencyClaim(request, signerOrProvider);
+  const contractAddress = erc20EscrowToPayArtifact.getAddress(request.currencyInfo.network!);
+
+  const signer = getSigner(signerOrProvider);
+  const tx = await signer.sendTransaction({
+    data: encodedTx,
+    to: contractAddress,
+    value: 0,
+    ...overrides,
+  });
+  return tx;
+}
+
+/**
+ * Processes a transaction to revertEmergencyClaim().
+ * @param request request to pay.
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function revertEmergencyClaim(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Web3Provider | Signer = getProvider(),
+  overrides?: ITransactionOverrides,
+): Promise<ContractTransaction> {
+  const encodedTx = encodeRevertEmergencyClaim(request, signerOrProvider);
+  const contractAddress = erc20EscrowToPayArtifact.getAddress(request.currencyInfo.network!);
+
+  const signer = getSigner(signerOrProvider);
+  const tx = await signer.sendTransaction({
+    data: encodedTx,
+    to: contractAddress,
+    value: 0,
+    ...overrides,
+  });
+  return tx;
+}
+
+/**
+ * Processes a transaction to refundFrozenFunds().
+ * @param request request to pay.
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function refundFrozenFunds(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Web3Provider | Signer = getProvider(),
+  overrides?: ITransactionOverrides,
+): Promise<ContractTransaction> {
+  const encodedTx = encodeRefundFrozenFunds(request, signerOrProvider);
+  const contractAddress = erc20EscrowToPayArtifact.getAddress(request.currencyInfo.network!);
+
+  const signer = getSigner(signerOrProvider);
+  const tx = await signer.sendTransaction({
+    data: encodedTx,
+    to: contractAddress,
+    value: 0,
+    ...overrides,
+  });
+  return tx;
+}
+
+/**
+ * Encodes the call to payEscrow().
+ * @param request request to pay.
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ */
+export function encodePayEscrow(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Web3Provider | Signer = getProvider(),
+  amount?: BigNumberish,
+  feeAmountOverride?: BigNumberish,
+): string {
+  validateRequest(request, PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT);
```

Comment:
I'm not yet super familiar with our Escrow solution... isn't the escrow a new Payment Network?

Reply:
No, it's using the erc20_fee_proxy.